source/extensions/common/wasm/wasm.cc

```diff
@@ -1230,16 +1236,33 @@ std::unique_ptr<Wasm> createWasm(absl::string_view id,
   return wasm;
 }
 
-std::shared_ptr<Wasm> createThreadLocalWasm(Wasm& base_wasm,
-                                            const envoy::config::wasm::v2::VmConfig& vm_config,
-                                            Event::Dispatcher& dispatcher,
-                                            absl::string_view configuration, Api::Api& api) {
+std::shared_ptr<Wasm> createWasm(absl::string_view id,
+                                 const envoy::config::wasm::v2::VmConfig& vm_config,
+                                 Upstream::ClusterManager& cluster_manager,
+                                 Event::Dispatcher& dispatcher, Api::Api& api) {
+  return std::shared_ptr<Wasm>(createWasmInternal(id, vm_config, cluster_manager, dispatcher, api));
```

Comment:
```
return std::shared_ptr<Wasm>(/* a rvalue unique_ptr can construct shared_ptr */ createWasmUnique(id, vm_config, cluster_manager, dispatcher, api));
```

Reply:
Yes, createWasmInternal now returns a unique_ptr.

Reply:
Now it returns shared_ptr as discussed below.